### PR TITLE
8208761: Update constant collections to use the new immutable collections

### PR DESCRIPTION
**Pull request by nlisker**
Resubmitting from before the transition to this repo. See linked issue for details.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8208761](https://bugs.openjdk.java.net/browse/JDK-8208761): Update constant collections to use the new immutable collections

### Reviewers
 * Ambarish Rapte ([arapte](@arapte) - **Reviewer**)
 * Kevin Rushforth ([kcr](@kevinrushforth) - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jfx pull/104/head:pull/104`
`$ git checkout pull/104`

### CODE REVIEW

**Comment by kevinrushforth** [comment]
Given the large number of changes, I'd like two reviewers to look it over for any possible typos.

File: modules/javafx.graphics/src/main/java/com/sun/prism/j2d/print/J2DPrinter.java [hunk 1]
```diff
@@ -629,60 +629,49 @@ public Paper defaultPaper() {
         return paperSet;
     }
 
-    private static Map<MediaTray, PaperSource> preDefinedTrayMap = null;
-    private static Map<MediaSizeName, Paper> predefinedPaperMap = null;
-    private static void initPrefinedMediaMaps() {
+    private static Map<MediaSizeName, Paper> predefinedPaperMap;
```
[the remark this answers is not in the document]
Comment (nlisker) [reply modules/javafx.graphics/src/main/java/com/sun/prism/j2d/print/J2DPrinter.java]: I think that "predefined" is correct as it's one word. I don't mind either way.

[the remark this answers is not in the document]
Comment (kevinrushforth) [reply modules/javafx.graphics/src/main/java/com/sun/prism/j2d/print/J2DPrinter.java]: @nlisker is right: `predefined` is one word.

File: modules/javafx.graphics/src/main/java/com/sun/prism/j2d/print/J2DPrinter.java [hunk 2]
```diff
@@ -629,60 +629,49 @@ public Paper defaultPaper() {
...
+    private static Map<MediaTray, PaperSource> preDefinedTrayMap;
 
+    private static void initPrefinedMediaMaps() {
```
[the remark this answers is not in the document]
Comment (kevinrushforth) [reply modules/javafx.graphics/src/main/java/com/sun/prism/j2d/print/J2DPrinter.java]: Yes, this should be fined, but with a lower-case `d`.

**Comment by kevinrushforth** [comment]
> The docs for an Unmodifiable Set, List, or Map state that they are "value-based", meaning that the identity of the stored objects cannot be relied upon.

I misread this. It is the Set, List or Map itself that is "value-based" not the entries contained therein. So there is no problem that I can see.

I will do more testing and a more careful review later.

**Comment by nlisker** [comment]
> The only comment I have is the one inline comment I left in KeyCodeMap.java about renaming the "put" method to something with "entry" in the name to reflect what it now does.

Will fix it after Ambarish reviews.

**Comment by arapte** [comment]
The fix looks good to me too, Shall approve once PR is updated for Kevin's comment.

**Pull request closed without merging**